### PR TITLE
CGPROD-2522 gelGrid param for showing select page

### PR DESCRIPTION
**Pull request by AnthonyGoodchild**
Adds gelGrid param for showing the page of a choice on load.

Used on select-screen via transientData

```
this.transientData["level-select"].choice = { id: "choice-id" };
```

https://jira.dev.bbc.co.uk/browse/CGPROD-2522

### CODE REVIEW

File: src/core/layout/grid/grid.js [hunk 1]
```diff
@@ -157,6 +158,13 @@ export class GelGrid extends Phaser.GameObjects.Container {
         return this._cells.filter((cell, idx) => idx >= pageMin && idx < pageMax);
     }
 
+    getCellPage(theme, cellId = 0) {
+        const cellIdx = theme.choices.reduce((cellIdx, cell, idx) => {
+            return cell.id === cellId ? idx : cellIdx;
+        }, 0);
```
Comment (FostUK) [inline src/core/layout/grid/grid.js:164]: Would findIndex work for this?
`const cellIdx = Math.Max(theme.choices.findIndex(cell => cell.id === cellId), 0);`

?

File: src/core/layout/grid/grid.js [hunk 2]
```diff
@@ -33,6 +33,7 @@ export class GelGrid extends Phaser.GameObjects.Container {
     }
 
     addGridCells(theme) {
+        this.page = this.getCellPage(theme, this._config.showChoice);
```
Comment (FostUK) [inline src/core/layout/grid/grid.js:36]: Any reason why you aren't using the `transientData.choice` value here that was getting saved anyway?

Comment (AnthonyGoodchild) [reply src/core/layout/grid/grid.js:36]: I've been over it again and while using `choice` does technically work, we would be saying that choice transient data does not necessarily reflect the choice which the player made.
 
If we are happy saying that then fine, I think it slightly muddies whatever the game does around the selected choice but any game changing this should be aware of what the value means anyway.

File: src/core/layout/grid/grid.js (hunk 1 above)
Comment (AnthonyGoodchild) [reply src/core/layout/grid/grid.js:164]: yeah not sure why I didn't do that

Comment (AnthonyGoodchild) [reply src/core/layout/grid/grid.js]: I guess it was the default value, I should use `Math.max` in these kinds of logic more often, very complimentary.

File: src/core/layout/grid/grid.js [hunk 3]
```diff
@@ -157,6 +158,14 @@ export class GelGrid extends Phaser.GameObjects.Container {
         return this._cells.filter((cell, idx) => idx >= pageMin && idx < pageMax);
     }
 
+    getCellPage(theme, cellId = 0) {
+        const cellIdx = Math.max(
+            theme.choices.findIndex(cell => cell.id === cellId),
+            0,
+        );
```
Comment (FostUK) [inline src/core/layout/grid/grid.js:165]: Aww! Prettier! So annoying :)

**Pull request merged**